src/com/exadel/bootcamp/CommandExecutor.java

```diff
@@ -3,80 +3,42 @@
         System.out.println("Welcome, enter you name");
         Scanner scanner = new Scanner(System.in);
         try {
             while (scanner.hasNextLine()){
                 String command = scanner.nextLine();
                 if (command.equals("quit")) {
                     scanner.close();
+                    return;
                 }
                 else if (command.equals("admin")) {
-                    runAdmin();
+                    runAdmin(scanner);
                 } else {
-                    runClient();
+                    runClient(scanner, command);
                 }
-
-                System.out.println("command set: " + command);
             }
         } catch(Exception e){
             System.out.println("Error ::"+e.getMessage());
             e.printStackTrace();
-        } finally {
-            if (scanner != null)
-                scanner.close();
         }
     }
-
-//    public static void identifyUser() {
-//        System.out.println("Enter you name");
-//        Scanner scanner = new Scanner(System.in);
-//        String inputValue = scanner.nextLine();
-//        System.out.println("value from console: " + inputValue);
-//
-//        if (inputValue.equals("admin")) {
-//            AdminUser admin = new AdminUser();
-//            System.out.println("Give a name to cash machine");
-//            Scanner adminScanner = new Scanner(System.in);
-//            String adminCommand = adminScanner.nextLine();
-//
-//            CashMachine c = new CashMachine("c1");
-//
-//            if (inputValue.equals("put")) {
-//                int banknote;
-//                int quantity;
-//
-//                System.out.println("Enter banknote size:");
-//                banknote = Integer.parseInt(inputValue);
-//                System.out.println("Enter quantity:");
-//                quantity = Integer.parseInt(inputValue);
-//
-//                admin.put(c, banknote, quantity);
-//                History.setHistory("put cash: " + banknote + " size - " + quantity + " quantity");
-//                System.out.println("Money was successfully loaded");
-//            } else if (inputValue.equals("home")) {
-//                identifyUser();
-//            }
-//        }
-//
-//        if (!Users.getUsers().keySet().contains(inputValue)) {
-//            Users.createUser(inputValue, new ClientAccount(inputValue));
-//        }
-//    }
-
-    public static void runAdmin() {
+    private static void runAdmin(Scanner scanner) {
         AdminUser admin = new AdminUser();
-        Scanner scanner = new Scanner(System.in);
-        System.out.println("What are you going to do...");
+        System.out.println("What are you going to do..?");
         try {
             while (scanner.hasNextLine()) {
                 String command = scanner.nextLine();
-                if (command.equals("exit")) {
-                    identifyUser();
+                if (command.equals("home")) {
```

Comment:
Please think of extracting code within `while` into a separate method.
Ideally, no method in Java should exceed 20-30 lines of code. This one is much longer)